### PR TITLE
#1692585870 a user or a manager should be able to comment on a trip requests

### PR DESCRIPTION
**Pull request by hezronkimutai**
#### What does this PR do?
-  This PR adds a feature that enables users and managers to comment on trip requests
-  This PR is also covers deletion of a comment.
#### Description of Task to be completed?
- This PR implements the comments feature
- Adds more styles to the single request page.
- Implements comment deletion
#### How should this be manually tested?
- `git clone https://github.com/andela/team-odd-bn-frontend`
- `git checkout ft-comment-on-trip-requests-1692585870`
- `npm run dev`
- navigate to `/requests` route
- Click on one request and try to create or delete a comment.
#### What are the relevant pivotal tracker stories?
[#169258587](https://www.pivotaltracker.com/n/projects/2407428/stories/169258587)
#### Screenshots (if appropriate)
Screen Shot 2020-01-22 at 12 06 28

Screen Shot 2020-01-23 at 12 35 51

Screen Shot 2020-01-23 at 10 48 19

### CODE REVIEW

File: src/helpers/errorProcessor.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+export default (error) => {
+  let err = error.response
+    ? error.response
+    : (error.response = { data: { message: 'Ooops! Network Error' } });
+  err = error.response.data.message
+    ? error.response.data.message
+    : error.response.data.error;
+  err = typeof err === 'object' ? err : [err];
+  return err;
+};
```
Comment (william000000) [inline src/helpers/errorProcessor.js:10]: Looks good!

File: src/views/trips/CommonSingleRequest.js [hunk 2]
```diff
@@ -2,74 +2,137 @@ import React, { Component } from 'react';
...
-                </div>
-              </div>
-            ))}
-            <div className="commentCardContainer">
-              <div className="commentCardHeader">
-                <div className="commentCardHeaderLeftSide">
-                    Comment
+                <div className="commentInputField">
+                  <textarea
+                    value={xx}
```
Comment (william000000) [inline src/views/trips/CommonSingleRequest.js]: use a meaningful name for this `xx` variable

**Review by william000000 — CHANGES_REQUESTED** [review]
Great Work @hezronkimutai, And remember to work on the pop box when you hit the delete button.

File: src/containers/TripsContainer/RoundTrip.js [hunk 3]
```diff
@@ -142,4 +142,4 @@ export const mapStateToProps = (state) => ({
 
 });
 
-export default connect(mapStateToProps, { createRoundTrip, updateSpinnerStatus, getTripAction })(RoundTrip);
+export default connect(mapStateToProps, { createRoundTrip, updateSpinnerStatus, getTripAction })(RoundTrip);
```
Comment (NiyongaboEric) [inline src/containers/TripsContainer/RoundTrip.js]: add EOL

**Review by NiyongaboEric — CHANGES_REQUESTED** [review]
Good job @hezronkimutai! I realized we need to do extra work to finish the work.

- I request delete the comment in real-time,  normally you would refresh the browser.
Screen Shot 2020-01-23 at 07 14 04

- What do you think if we design custom modal or find packages to use instead of using browser alert
Screen Shot 2020-01-23 at 07 22 42

File: src/assets/css/table.scss [hunk 4]
```diff
@@ -6,7 +6,7 @@ $contents: (tripId: 'Trip Id ',
   status: 'status',
   action: 'action',
 );
-$tt: (originId: 'originId',
+$tt: (originId: 'origin',
   destinationId: 'destination',
   startDate: 'start date',
   returnDate: 'return date',
```
Comment (victkarangwa) [inline src/assets/css/table.scss]: Could you please write the headers in sentence case?
like this:
```js
$tt: (originId: 'Origin',
destinationId: 'Destination',
startDate: 'Start date',
 returnDate: 'Return date',
  reason: 'Eeason',
```

**Review by NiyongaboEric — APPROVED** [review]

**Review by victkarangwa — COMMENTED** [review]
Good job @hezronkimutai 
Take a look at the feedback I left!

**Review by victkarangwa — APPROVED** [review]
Good job @hezronkimutai

**Review by shemaeric — APPROVED** [review]
Good work @hezronkimutai

**Pull request merged**